### PR TITLE
PCCR V1 | Deprecate task

### PR DESCRIPTION
**Pull request by chandrashritii**
**Task name**: Publish Code Coverage Results V1

**Description**: 

This PR will add a deprecation warning to PCCRV1 task asking the user to switch to PCCRV2.

- Added a deprecation warning to PCCR V1 task
- Bumped up task version from 2.228.0 to 2.235.0

**Documentation changes required:** (Y/N) Y

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

image

image

### CODE REVIEW

File: Tasks/PublishCodeCoverageResultsV1/publishcodecoverageresults.ts [hunk 1]
```diff
@@ -30,6 +30,9 @@ async function run() {
             tempFolder = resolvePathToSingleItem(workingDirectory, reportDirectory, true);
         }
 
+        //Warning for V1 deprecation
```
Comment (triptijain2112) [inline Tasks/PublishCodeCoverageResultsV1/publishcodecoverageresults.ts:33]: Can we have this warning on line  no 10, just at the beginning of run method before any logs are printed?

File: Tasks/PublishCodeCoverageResultsV1/task.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "id": "2A7EBC54-C13E-490E-81A5-D7561AB7CD97",
   "name": "PublishCodeCoverageResults",
   "friendlyName": "Publish code coverage results",
-  "description": "Publish Cobertura or JaCoCo code coverage results from a build",
+  "description": "[DEPRECATION WARNING. Users are recommended to switch to version 2.235.0.] Publish Cobertura or JaCoCo code coverage results from a build",
```
Comment (vinayakmsft) [inline Tasks/PublishCodeCoverageResultsV1/task.json:5]: I think the exact minor version would keep on changing in the future. Should we just say , Switch to Version 2.

**Review by vinayakmsft — COMMENTED** [review]
test comment

File: Tasks/PublishCodeCoverageResultsV1/task.json (hunk 2 above)
Comment (triptijain2112) [reply Tasks/PublishCodeCoverageResultsV1/task.json:5]: +1
switch to major version 2

File: Tasks/PublishCodeCoverageResultsV1/task.json [hunk 3]
```diff
@@ -99,6 +99,7 @@
     "FailedToGenerateHtmlReport": "Failed to generate Html report. Error: %s",
     "IgnoringReportDirectory": "Ignoring coverage report directory with Html content as we are auto-generating Html content",
     "UpgradeAgentMessage": "Please upgrade your agent version. https://github.com/Microsoft/vsts-agent/releases",
-    "GeneratedHtmlReport": "Generated code coverage html report: %s"
+    "GeneratedHtmlReport": "Generated code coverage html report: %s",
+    "V1TaskDeprecationNotice": "Version 1.234.0 for Publish Code Coverage Results task will be deprecated in 2024. Users are strongly recommended to switch to version 2.235.0"
```
Comment (rasunkar) [inline Tasks/PublishCodeCoverageResultsV1/task.json:103]: should we add a link to the public documentation + blog

File: Tasks/PublishCodeCoverageResultsV1/publishcodecoverageresults.ts (hunk 1 above)
Comment (rasunkar) [reply Tasks/PublishCodeCoverageResultsV1/publishcodecoverageresults.ts:33]: would this visual represent to the user when they plan to add it in classic definition? Please check on how we can show this in yml definition as well.

**Pull request closed without merging**